app/schema_validation/definitions.py

```diff
@@ -19,14 +19,44 @@
     "link": "link to our error documentation not yet implemented"
 }
 
-
 personalisation = {
     "type": "object",
-    "code": "1001",  # yet to be implemented
-    "link": "link to our error documentation not yet implemented"
+    "patternProperties": {
+        "^.*$": {
+            "type:": "object",
+            "properties": {
+                "file": {
+                    "type": "string",
+                    "binaryEncoding": "base64",
+                },
+                "sending_method": {
+                    "type": "string",
+                    "enum": ["attach", "link"]
+                }
+            },
+            "required": [
+                "file", "sending_method"
+            ],
+            "if": {
+                "properties": {
+                    "sending_method": {
+                        "const": "attach"
+                    }
+                }
+            },
+            "then": {
+                "required": ["filename"],
+                "properties": {
+                    "filename": {
+                        "minLength": 3,
+                        "maxLength": 255
+                    }
+                }
+            }
```

Comment:
Good stuff. It reminds me a bit of XHTML2 (XForms & cie) and/or XSLT with the declarative conditional.